File: server-side/src/main/java/pl/agh/io/PlannedWorkout.java
package pl.agh.io;

import pl.agh.io.exercise.models.Exercise;
import pl.agh.io.gym.models.Gym;

import java.util.List;

public class PlannedWorkout {
    private final List<Gym> gymsToVisit;
    private final List<List<Exercise>> exercisesOnGyms;

    public PlannedWorkout(List<Gym> gymsToVisit, List<List<Exercise>> exercisesOnGyms) {
        this.gymsToVisit = gymsToVisit;
        this.exercisesOnGyms = exercisesOnGyms;
    }

    public List<Gym> getGymsToVisit() {
        return gymsToVisit;
    }

    public List<List<Exercise>> getExercisesOnGyms() {
        return exercisesOnGyms;
    }
}

File: lib/models/planned_workout.dart
import 'exercise.dart';
import 'gym.dart';
import '../utils/route.dart' as Route;

class PlannedWorkout {
  final List<Gym> gymsToVisit;
  final List<List<Exercise>> exercisesOnGyms;
  final Route.Route route;

  PlannedWorkout({required this.gymsToVisit, required this.exercisesOnGyms, required this.route});

  @override
  String toString() {
    return 'PlannedWorkout{gymsToVisit: $gymsToVisit, exercisesOnGyms: $exercisesOnGyms}';
  }
}